examples/nlp/semantic_similarity_with_bert.py

```diff
@@ -0,0 +1,362 @@
+
+"""
+Distribution of our training targets.
+"""
+print("Train Target Distribution")
+print(train_df.similarity.value_counts())
+
+"""
+Distribution of our validation targets.
+"""
+print("Validation Target Distribution")
+print(valid_df.similarity.value_counts())
+"""
+ We have some "-" in our train and validation targets, we will not use those.
+"""
+train_df = (
+    train_df[train_df.similarity != "-"]
+    .sample(frac=1.0, random_state=42)
+    .reset_index(drop=True)
+)
+valid_df = (
+    valid_df[valid_df.similarity != "-"]
+    .sample(frac=1.0, random_state=42)
+    .reset_index(drop=True)
+)
+
+"""
+One hot encoding training, validation and test labels
+"""
+train_df["label"] = train_df["similarity"].apply(
+    lambda x: 0 if x == "contradiction" else 1 if x == "entailment" else 2
+)
+y_train = tf.keras.utils.to_categorical(train_df.label, num_classes=3)
+
+valid_df["label"] = valid_df["similarity"].apply(
+    lambda x: 0 if x == "contradiction" else 1 if x == "entailment" else 2
+)
+y_val = tf.keras.utils.to_categorical(valid_df.label, num_classes=3)
+
+test_df["label"] = test_df["similarity"].apply(
+    lambda x: 0 if x == "contradiction" else 1 if x == "entailment" else 2
+)
+y_test = tf.keras.utils.to_categorical(test_df.label, num_classes=3)
+
+"""
+## Keras Custom Data Generator
+"""
+
+
+class BertSemanticDataGenerator(tf.keras.utils.Sequence):
+    """
+        Generates batch data for Keras.
+
+        Args:
+            sentence1:          Array of premise input sentences.
+            sentence2:          Array of the hypothesis input sentences.
+            labels:             Array of labels.
+            batch_size:         Integer size of the batch.
+            shuffle:            boolean shuffle data or not.
+            include_targets:    boolean generate data for train/test purpose.
+        Returns:
+            encoded_features:   input_ids,
+                                attention_mask,
+                                token_type_ids,
+                                labels if include_targets is set to true.
+    """
+
+    def __init__(
+        self,
+        sentence1,
+        sentence2,
+        labels,
+        batch_size=batch_size,
+        shuffle=True,
+        include_targets=True,
+    ):
+        self.sentence1 = sentence1
+        self.sentence2 = sentence2
+        self.labels = labels
+        self.shuffle = shuffle
+        self.batch_size = batch_size
+        self.include_targets = include_targets
+        # Load our BERT Tokenizer to encode the text.
+        # We will use base-base-uncased pretrained model.
+        self.tokenizer = transformers.BertTokenizer.from_pretrained(
+            "bert-base-uncased", do_lower_case=True
+        )
+        self.on_epoch_end()
+
+    def __len__(self):
+        # Denotes the number of batches per epoch.
+        return len(self.sentence1) // self.batch_size
+
+    def __getitem__(self, idx):
+        # Generates batch of data.
```

Comment:
You can be more precise: "Retrieves the batch of index `idx`."

Reply:
Added